### PR TITLE
tighten up the invalidation story for fullselfcacher

### PR DESCRIPTION
**Pull request by maxtaco**
- allow for a 10min timeout in a worst case scenario
- add tests
- allow UPAKCache and FullSelfCacher to cross-polinate each other

### CODE REVIEW

File: go/engine/upak_loader_test.go [hunk 1]
```diff
@@ -150,4 +166,113 @@ func TestLoadDeviceKeyRevoked(t *testing.T) {
...
+	// user's sigchain should stay the same.
+	tc.G.GetFullSelfer().WithSelf(context.TODO(), func(u *libkb.User) error {
+		require.NotNil(t, u)
+		require.True(t, u.GetSigChainLastKnownSeqno() == scv)
+		return nil
+	})
+
+	// After the CachedUserTimeout, the FullSelfer ought to repoll.
+	// Check that the sigchain is updated after the repoll, which reflects
+	// the new dvice having been added.
```
Comment (patrickxb) [inline go/engine/upak_loader_test.go]: device

File: go/libkb/full_self_cacher.go [hunk 2]
```diff
@@ -139,6 +188,41 @@ func (m *CachedFullSelf) WithUser(arg LoadUserArg, f func(u *User) error) (err e
 	return f(u)
 }
 
+func (m *CachedFullSelf) cacheMe(u *User) {
+	m.me = u
+	m.cachedAt = m.G().Clock().Now()
+}
+
+// Update updates the CachedFullSelf with a User loaded from somplace else -- let's
```
Comment (patrickxb) [inline go/libkb/full_self_cacher.go]: someplace

File: go/libkb/upak_loader.go [hunk 3]
```diff
@@ -171,9 +171,22 @@ func (u *CachedUPAKLoader) putUPAKToCache(ctx context.Context, obj *keybase1.Use
 
 	uid := obj.Base.Uid
 	u.G().Log.CDebugf(ctx, "| Caching UPAK for %s", uid)
+
```
Comment (patrickxb) [inline go/libkb/upak_loader.go]: this looks good!

**Comment by maxtaco** [comment]
thanks for the review @patrickxb !

**Pull request merged**